build/chip/java/rules.gni

```diff
@@ -176,6 +176,161 @@ template("java_library") {
   }
 }
 
+# Declare a java binary target
+#
+#   sources: List of .java files included in this binary. Mutually exclusive with jar_path.
+#
+#   jar_path: A path to an existing JAR. Mutually exclusive with sources.
+#
+#   output_name: File name for the output .jar (not including extension).
+#     Defaults to the input .jar file name.
+#
+#   javac_flags: additional flags to pass to the javac compiler
+#
+template("java_binary") {
```

Comment:
This seems like it is 99% similar to the `java_library` template. Can you possibly templat-ize this common bit of logic and have both the `java_binary` and `java_library` re-use that?

Reply:
I tried without success, still working on it, I see chromium GN rules also do something like this, so I need to check if we can call an template from another template in GN

Reply:
Create https://github.com/project-chip/connectedhomeip/issues/23080 to investigate it